plugins/tasks/tasks-core/src/com/intellij/tasks/gitlab/GitlabRepository.java

```diff
@@ -300,4 +310,74 @@ private HttpGet getApiVersionRequest() {
   public void setProjects(@NotNull List<GitlabProject> projects) {
     myProjects = projects;
   }
+
+
+  @Override
+  protected int getFeatures() {
+    return super.getFeatures() | TIME_MANAGEMENT;
+  }
+
+  /**
+   * Adds time spent to a task.
+   *
+   * @param task      The local task we are submitting time for.
+   * @param timeSpent The amount of time spent on the issue in the format 0h0m.
+   * @param comment   The comment to also add to the issue.
+   * @throws Exception
+   */
+  @Override
+  public void updateTimeSpent(@NotNull LocalTask task, @NotNull String timeSpent, @NotNull String comment) throws Exception {
+    final Pattern issueURLPattern = Pattern.compile("https://.*?/(.*?/.*?)/.*"); // Captures project namespace from URL
```

Comment:
This is fragile and won't work with Gitlab instances hosted locally (no HTTPS, custom root paths for the service like `http://localhost/tracker/user/project/issues/42`, etc.). I'd at least bind it to `issue/:iid:` at the end (e.g. `https?://.*/([^/]*/[^/]*)/issues/\d+`).